secure.c

```diff
@@ -440,15 +441,25 @@ sec_out_mcs_data(STREAM s, uint32 selected_protocol)
 	out_uint32_le(s, g_keyboard_functionkeys);
 	out_uint8s(s, 64);	/* reserved? 4 + 12 doublewords */
 	out_uint16_le(s, 0xca01);	/* colour depth? */
-	out_uint16_le(s, 1);
+	out_uint16_le(s, 1);		/* product id */
 
-	out_uint32(s, 0);
-	out_uint8(s, g_server_depth);
-	out_uint16_le(s, 0x0700);
+	out_uint32(s, 0);		/* serial number */
+	out_uint8(s, g_server_depth);	/* this stuff is misaligned should be 2,2,2,64,1,1 */
+	out_uint16_le(s, 0x0700);	/* this strange constant is split across fields */
 	out_uint8(s, 0);
 	out_uint32_le(s, 1);
 	out_uint8s(s, 64);
 	out_uint32_le(s, selected_protocol);	/* End of client info */
+	if (g_dpi > 0)
+	{
+		/* Extended client info describing monitor geometry */
+		out_uint32_le(s, g_width * 254 / (g_dpi * 10)); /* desktop physical width */
+		out_uint32_le(s, g_height * 254 / (g_dpi * 10)); /* desktop physical height */
+		out_uint16_le(s, 0); /* orientation: landscape */
+		out_uint32_le(s, g_dpi < 96 ? 100 : (g_dpi * 100 + 48) / 96); /* desktop scale factor */
```

Comment:
Maybe add a comment about desktopScaleFactor bounds (100%-500%) too?
Or I just a little bit picky?:)

Reply:
I don't believe the upper bound is accurate in the spec.